src/client/automation/playback/type/type-text.js

```diff
@@ -31,7 +31,9 @@ function _updateSelectionAfterDeletionContent (element, selection) {
     var startNode      = selection.startPos.node;
     var hasStartParent = startNode.parentNode && startNode.parentElement;
 
-    if (browserUtils.isWebKit || !hasStartParent || !domUtils.isElementContainsNode(element, startNode)) {
+    var browserShouldUpdateSelection = browserUtils.isChrome && browserUtils.version < 58 || browserUtils.isSafari;
```

Comment:
`browserShouldUpdateSelection` - from the var's name it seems that the browser should update selection itself and we shouldn't do it. But I suppose it should be `browserRequiresSelectionUpdating`. Am I right? /cc @georgiy-abbasov